lua/controller.lua

```diff
@@ -273,11 +274,18 @@ local function controller_insert_to_drawers(pos, stack)
 		-- store, the drawer is not full, and the drawer entity is loaded, we
 		-- will put the items in the drawer
 		if content.name == stack:get_name() and
-				content.count < content.maxCount and
 				drawers.drawer_visuals[core.hash_node_position(drawer_pos)] then
-			return drawers.drawer_insert_object(drawer_pos, stack, visualid)
+			if content.count < content.maxCount then
+				return drawers.drawer_insert_object(drawer_pos, stack, visualid)
+			else
+				new_drawer = true
```

Comment:
If the drawer is full, then we fall into the "empty" section below, now.

Reply:
I'm sorry but with this new `new_drawer` variable the code is hardly readable. Please think of a better logic or maybe do some refactoring so a better logic is possible. :)